### DISCUSSION TITLE
Enable/disable inverters based on power need

### DISCUSSION THREAD

**Discussion by hofvanta** [ksheumaker/homeassistant-apsystems_ecur#232] [Ideas]
Is it an idea since the switch is added to switch inverters on/of based on electricity prices to group inverter ID based on roof position and use sun azimuth to disable some of the inverters depends on the power usage? More and more energy providers (for example in NL) charge house holds with solar installations per month in the summer when the prices drops negative.
Most HA integration and solar users know pretty well want the current power consumption is and you can enable/disable inverters based on the energy consumption (automation). 
As you group them as part of APSystemes-ECUR integration, know which ones on which sides can be switches off, to sync the amount of inverters to the amount of consumption need. 

(I love the integration and working well. Keep up the good working :-) )

**Comment by HAEdwin** [comment]
I understand the need and it would be nice to have that advanced automation. I think this feature would take the integration to a whole different level, but would require dependencies on other integrations. Maybe another programmer will want to take this over, but unfortunately I won't - it's just too much, not generic work. There could be a solution in creating HA automation scripts, I don't know. It's a good idea.

**Comment by hofvanta** [comment]
I understand. Is a roof and per inverter enable-/disablement something what will fit in the integration? Then I can make a blueprint for the rest of the automation. Otherwise it will be a complete new integration to send HTTP calls to disable/enable one or more inverters specially for APsystems.

**Reply by HAEdwin** [reply]
Switching on/off per inverter should be possible. The integration then generates multiple switches that you can group yourself in an automation. I just have to find the time to adjust it, it might take a while.

**Reply by hofvanta** [reply]
No problem. As you know also the inverter IDs, can grouping then a part of APSystems-ECUR? Then you can use the sun integration (azimuth) and the automation to control everything with automations.

**Reply by HAEdwin** [reply]
Yes, switches would be dynamically created and linked to the inverter-ID. Azimut can be retreived externally via automation script. Strange thing it cannot be retreived within the integration but that's a core design issue.

**Comment by wirecuttr** [comment]
I was wondering if it's possible to do the same. If the integration can provide a switch per inverter, that would be great. The rest I think  would be reasonable  to handle outside the integration as individual wants and needs will vary

**Comment by hofvanta** [comment]
I now have a stable solution and a switch per inverter.
Anyone interested in helping to devise a workable automation system to dynamically switch inverters based on energy need?

To create a switch per inverter
Switch template YAML:
```
- platform: template
  switches:
    inverter_<inverter_ID>:
      friendly_name: "inverter_<inverter_ID>"
      turn_on:
        action: rest_command.inverter_<inverter_ID>_on
      turn_off:
        action: rest_command.inverter_<inverter_ID>_off
```

Rest-command yaml
```
inverter_<inverter_ID>_on:
  url: !secret apsystems_url_inverter_set_state
  method: POST
  content_type: "application/x-www-form-urlencoded"
  payload: "ids[]=<inverter_ID>1"
inverter_<inverter_ID>_off:
  url: !secret apsystems_url_inverter_set_state
  method: POST
  content_type: "application/x-www-form-urlencoded"
  payload: "ids[]=<inverter_ID>2"
```

**Reply by HAEdwin** [reply]
I've also implemented the inverter switch in a revised version of the integration but I have to do some further testing on it before releasing. 

Three weeks ago my old ECU broke down (I thought) so I bought a new one starting with 2162. Now I can test both models, which is kind of reassuring when releasing the new version. Before this revised version I worked on the proxy version so it all took a while but it's kind off cool to have the ability to turn individual inverters on and off, before the summer starts, using automations. Thank you for supplying the URL's btw. and sorry that it took so long before implementing.

I'll try to make the repository available as soon as possible so that there's a AIO solution.
